packages/@angular/cli/models/webpack-configs/common.ts

```diff
@@ -67,7 +67,7 @@ export function getCommonConfig(wco: WebpackConfigOptions) {
     devtool: buildOptions.sourcemaps ? 'source-map' : false,
     resolve: {
       extensions: ['.ts', '.js'],
-      modules: [nodeModules],
+      modules: ['node_modules', nodeModules],
```

Comment:
I get it now, it's for when the linked library does not contain the peerDep but the app does. 👍